src/bokeh/embed/bundle.py

```diff
@@ -197,28 +179,25 @@ def bundle_for_objs_and_resources(objs: Sequence[Model | Document] | None,
 
     from copy import deepcopy
 
-    if js_resources:
-        js_resources = deepcopy(js_resources)
-        if not use_widgets and "bokeh-widgets" in js_resources.js_components:
-            js_resources.js_components.remove("bokeh-widgets")
-        if not use_tables and "bokeh-tables" in js_resources.js_components:
-            js_resources.js_components.remove("bokeh-tables")
-        if not use_gl and "bokeh-gl" in js_resources.js_components:
-            js_resources.js_components.remove("bokeh-gl")
-        if not use_mathjax and "bokeh-mathjax" in js_resources.js_components:
-            js_resources.js_components.remove("bokeh-mathjax")
-
-        js_files.extend(js_resources.js_files)
-        js_raw.extend(js_resources.js_raw)
-
-    if css_resources:
-        css_resources = deepcopy(css_resources)
-        css_files.extend(css_resources.css_files)
-        css_raw.extend(css_resources.css_raw)
-
-    if js_resources:
-        extensions = _bundle_extensions(all_objs if objs else None, js_resources)
-        mode = js_resources.mode if resources is not None else "inline"
+    if resources is not None:
+        resources = deepcopy(resources)
```

Comment:
I know this is how it was previously, but I guess a `clone` method with tests would be a little more confidence inspiring. It's just easy to imagine we unintentionally introduce something that is not handled well by default `deepcopy`

Reply:
I implemented `Resources.clone()` and refactored this code.